reasonml/data_structures/stack.re

```diff
@@ -0,0 +1,26 @@
+type stack('a) = list('a);
+
+let make = () => [];
+
+let push = (s: stack('a), e: 'a) => [e, ...s]
+
+let pop = (s: stack('a)) => {
+  switch s {
+    | [_, ...tl] => [...tl]
+    | _ => [];
+  }
+}
+
+let top = (s: stack('a)) => {
+  switch s {
+    | [] => None;
+    | [hd, ...tl] => Some(hd);
```

Comment:
Wouldn't `[hd, _]` be better?
Does the current code work if the stack contains only 1 element?